### ISSUE TITLE
SuppressWarnings for ExcessivePublicCount

### ISSUE THREAD

**Issue by juliangut** [phpmd/phpmd#409]
Warning suppression for [ExcessivePublicCount](https://phpmd.org/rules/codesize.html#excessivepubliccount) doesn't seem to be working properly at least for static methods

```php
/**
 * @SuppressWarnings(PHPMD.ExcessivePublicCount)
 */
class IntensivePublicStaticMethodClass
{
    public static function aMethod() {}
    public static function bMethod() {}
    public static function cMethod() {}
    public static function dMethod() {}
    public static function eMethod() {}
    ...
}
```

**Comment by ravage84** [comment]
Have you tried ``TooManyPublicMethods`` instead?

**Comment by juliangut** [comment]
Indeed it works @ravage84 

Anyway shouldn't ExcessivePublicCount work as well? Maybe not in my case as this class only has public static methods and no attribute at all?

**Comment by ravage84** [comment]
I guess it should and suppression should work, too.

Can you try to suppress everthing for testing?

`````php
/**
 * @SuppressWarnings(PHPMD)
 */
````

**Comment by juliangut** [comment]
I tried and `@SuppressWarnings(PHPMD)` does suppress the warning

**Comment by ravage84** [comment]
Odd, either the whole suppression does not work for you or something else is broken.
Can you try to replicate it on another machine etc.?

Try to suppress some other case, e.g. short variable name.

**Comment by juliangut** [comment]
I've made a quick test class with lots of code smells to test the all this suppressions

```php
/**
 * @SuppressWarnings(PHPMD)
 * @SuppressWarnings(PHPMD.UnusedPrivateField)
 * @SuppressWarnings(PHPMD.UnusedLocalVariable)
 * @SuppressWarnings(PHPMD.UnusedPrivateMethod)
 * @SuppressWarnings(PHPMD.UnusedFormalParameter)
 * @SuppressWarnings(PHPMD.ShortVariable)
 * @SuppressWarnings(PHPMD.LongVariable)
 * @SuppressWarnings(PHPMD.ShortMethodName)
 * @SuppressWarnings(PHPMD.ConstantNamingConventions)
 * @SuppressWarnings(PHPMD.BooleanGetMethodName)
 * @SuppressWarnings(PHPMD.ExitExpression)
 * @SuppressWarnings(PHPMD.EvalExpression)
 * @SuppressWarnings(PHPMD.Superglobals)
 * @SuppressWarnings(PHPMD.CamelCaseClassName)
 * @SuppressWarnings(PHPMD.CamelCasePropertyName)
 * @SuppressWarnings(PHPMD.CamelCaseMethodName)
 * @SuppressWarnings(PHPMD.CamelCaseParameterName)
 * @SuppressWarnings(PHPMD.CamelCaseVariableName)
 * @SuppressWarnings(PHPMD.CyclomaticComplexity)
 * @SuppressWarnings(PHPMD.NPathComplexity)
 * @SuppressWarnings(PHPMD.ExcessiveParameterList)
 * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
 * @SuppressWarnings(PHPMD.ExcessiveClassLength)
 * @SuppressWarnings(PHPMD.ExcessivePublicCount)
 * @SuppressWarnings(PHPMD.TooManyFields)
 * @SuppressWarnings(PHPMD.TooManyMethods)
 * @SuppressWarnings(PHPMD.TooManyPublicMethods)
 * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
 */
```

Among all of them the only one that doesn't suppress the warning is `@SuppressWarnings(PHPMD.ExcessivePublicCount)`, don't think it has anything to do with my machine

**Comment by juliangut** [comment]
An update. I've just tried with another project in several different machines and definitively `@SuppressWarnings(PHPMD.ExcessivePublicCount)` is not working

**Comment by ravage84** [comment]
Thanks for the feedback.
Tried to track down the suppression code by looking at the classes. Didn't find anything.
I guess I will need to debug it.
Not sure when I have time for that.

**Comment by ravage84** [comment]
Similar but not completyl same issue was mentioned in #493
